cancer_ai/chain_models_store.py

```diff
@@ -114,3 +116,112 @@ async def retrieve_model_metadata(self, hotkey: str) -> Optional[ChainMinerModel
         # The block id at which the metadata is stored
         model.block = metadata["block"]
         return model
+
+class ModelInfoTable(Base):
+    __tablename__ = 'models'
+    competition_id = Column(Integer, nullable=False)
+    hf_repo_id = Column(String, nullable=False)
+    hf_model_filename = Column(String, nullable=False)
+    hf_repo_type = Column(String, nullable=False)
+    hf_code_filename = Column(String, nullable=False)
+    date_uploaded = Column(DateTime, nullable=False)
+    block = Column(Integer, nullable=False)
+    hotkey = Column(String, nullable=False)
+
+    __table_args__ = (
+        PrimaryKeyConstraint('date_uploaded', 'hotkey', name='pk_date_hotkey'),
+    )
```

Comment:
I'm not seeing the need for such generalization right now since its used only once both ways. In future if we will have a need to introduce conversion more then once then it might be a good idea